src/leetCodeExplorer.ts

```diff
@@ -64,57 +73,100 @@ export class LeetCodeTreeDataProvider implements vscode.TreeDataProvider<LeetCod
+                }), false),
             ];
         }
-        if (!element) {
+        if (!element) { // Root view
             return new Promise(async (resolve: (res: LeetCodeNode[]) => void): Promise<void> => {
                 await this.getProblemData();
-                resolve(this.composeDifficultyNodes());
+                resolve([
+                    new LeetCodeNode(Object.assign({}, list.IProblemDefault, {
+                        id: "Root",
+                        name: "Difficulty",
+                    }), false),
+                    new LeetCodeNode(Object.assign({}, list.IProblemDefault, {
+                        id: "Root",
+                        name: "Tag",
+                    }), false),
+                    new LeetCodeNode(Object.assign({}, list.IProblemDefault, {
+                        id: "Root",
+                        name: "Company",
+                    }), false),
+                    new LeetCodeNode(Object.assign({}, list.IProblemDefault, {
+                        id: "Root",
+                        name: "Favorite",
+                    }), false),
+                ]);
             });
         } else {
-            return element.isProblem ? [] : this.composeProblemNodes(element.name);
+            switch (element.name) { // First-level
+                case "Difficulty":
+                case "Tag":
+                case "Company":
+                    return this.composeCategoryNodes(element);
+                case "Favorite":
+                    return this.treeData.Favorite.map((p: list.IProblem) => new LeetCodeNode(p));
+                default: // Second and lower levels
+                    return element.isProblem ? [] : this.composeProblemNodes(element);
+            }
         }
     }
 
     private async getProblemData(): Promise<void> {
-        const allProblems: list.IProblem[] = await list.listProblems();
-        this.treeData.clear();
-        for (const problem of allProblems) {
-            const problems: list.IProblem[] | undefined = this.treeData.get(problem.difficulty);
-            if (problems) {
-                problems.push(problem);
-            } else {
-                this.treeData.set(problem.difficulty, [problem]);
+        this.treeData = {
+            Difficulty: new Map(),
+            Tag: new Map(),
+            Company: new Map(),
+            Favorite: [],
+        };
+        for (const problem of await list.listProblems()) {
+            // Add favorite problem, no matter whether it is solved.
+            if (problem.favorite) {
+                this.treeData.Favorite.push(problem);
+            }
+            // Hide solved problem in category view is specified in option.
+            if (problem.state === ProblemState.AC && this.leetCodeConfig.get<boolean>("hideSolved")) {
+                continue;
+            }
+            // Add problems according to category
+            const categories: Array<[Category, string[]]> = [
```

Comment:
Can we use Map here?

use index to represent the `parent-children` relationship is confusing

Reply:
We could use Map here this way:
```typescript
const categories: Map<Category, string[]> = new Map<Category, string[]>([
    ["Difficulty", [problem.difficulty]],
    ["Tag", problem.tags],
    ["Company", problem.companies],
]);
```
Btw, a better way would be to use `Object.entries` here, but it is not supported by es6 in the `tsconfig.json`.

Reply:
Let's use Map here.